### PR TITLE
GitHub actions build

### PR DESCRIPTION
**Pull request by jhenshall**
Work in progress! First attempt at building site with GitHub Actions. Creating PR for discussion.

### CODE REVIEW

File: .github/workflows/gulp-build.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Install dependencies
+      run: npm install
+    - name: Compile site to docs folder
+      run: gulp build
+
+    - name: Commit changes
+      uses: EndBug/add-and-commit@v4
+      with:
+        message: "Site deployed"
+        add: "/docs"
```
Comment (jhenshall) [inline .github/workflows/gulp-build.yml]: I'm not sure this does what I intend it to.

**Comment by jhenshall** [comment]
Changed to gulp build (thank you @Hagellach37) and added bit to commit changes - not sure this does what I intend. Should maybe instead use [this example](https://github.com/marketplace/actions/github-pages-action) and `publish_dir: ./docs` rather than `add: "/docs"`?

File: .github/workflows/gulp-build.yml (hunk 1 above)
Comment (laurentS) [reply .github/workflows/gulp-build.yml]: If you're able to deploy without adding a commit to your history, I would tend to do that instead. In reality, I don't think you care about storing the compiled version of the pages, as it just messes up the commit history.

**Comment by laurentS** [comment]
About testing this, I think one way to do it would be:
1. remove the constraint that the action only runs on `master`, to allow it to run on this branch
2. commit/push to run the action/repeat until it works
If things go pear shaped, just redeploy from master like you did so far

File: .github/workflows/gulp-build.yml (hunk 1 above)
Comment (jhenshall) [reply .github/workflows/gulp-build.yml]: Agree that sounds much cleaner. Would the other example you sent achieve that?

**Comment by Hagellach37** [comment]
just looking at all those commits last night: did it finally work as expected?

In general I'm a bit confused that some files seem to be at so many places at the same time. For example there are:
* `index.html`
* `index.shtml`
* `docs/index.html`

**Comment by jhenshall** [comment]
Hey @Hagellach37 - nope, had to concede defeat on this. We need to roll back the last two commits on master because it's made a total mess.

**Comment by jhenshall** [comment]
I think I finally got there! A final once over, merge to master and then check it is all working as expected for prod.

File: .github/workflows/gulp-build.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (laurentS) [inline .github/workflows/gulp-build.yml:11]: A bit of nitpicking: if there's a way to replace `ubuntu-latest` with a specific version (whatever "latest" is now https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on looks like `ubuntu-18.04`), I would do that. It is called "dependency pinning" and prevents a lot of potential problems in the future. With "latest", you're basically saying "give me any version you decide on", remember what happened with the wrong version of node... If you pin the version, then you are in control of when to change that, and you can make sure it works before switching. Without pinning, it's likely that the change will happen without you noticing, and of course you will find out just as you deploy and are about to go do something else.... and now your site is broken :)

**Review by laurentS — APPROVED** [review]
Looks good to me! I've left a comment that is not a blocking point, but maybe worth changing just after you've merged the PR.
Also, if there are parts of this script that took a while to figure out and are not obvious, I'd suggest leaving a comment in here, just to explain why the non obvious thing is done this way, it'll help further down the line, when you or someone else needs to come back to it. If you don't want to pollute the script, you could also make some notes in the related ticket, for the same reasons.
Well done!

**Comment by jhenshall** [comment]
Thanks for feedback @laurentS ! Good points. Will merge now and come back to these.

**Pull request merged**